### PR TITLE
RavenDB-13267 - Reduce the CPU usage of GetSizeOnDisk

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-13267

### CODE REVIEW

**Comment by redknightlois** [comment]
That does take out the `TimeoutManager` and `FetchDatabase` usage of `GenerateReport`?

**Comment by grisha-kotler** [comment]
`TimeoutManager`?
The only place that now uses `StorageEnvironment.GenerateReport` is `debug/storage/report` (studio view -> Storage Report).

**Comment by redknightlois** [comment]
Excellent... for reference this was the call chain from `TimeoutManager`
image

**Comment by grisha-kotler** [comment]
Makes sense. This is the server dashboard.

**Comment by grisha-kotler** [comment]
What was the most expensive call inside `GenerateReport`? TreeIterator?

**Comment by redknightlois** [comment]
THE WHOLE THING, but certainly the tree iteration gets the lions' share

image

File: src/Voron/StorageEnvironment.cs [hunk 1]
```diff
@@ -875,9 +875,53 @@ internal void TransactionCompleted(LowLevelTransaction tx)
             }
         }
 
+        public SizeReport GenerateSizeReport()
+        {
+            long journalsSize = 0;
+            foreach (var journal in Journal.Files.ToList())
```
Comment (ayende) [inline src/Voron/StorageEnvironment.cs]: Why do we have the allocation on `ToList` here?  The `Files` is immutable.

**Review by ayende — COMMENTED** [review]
I'm not seeing anything here that should dramatically reduce the CPU usage. 
What is the hot spot that was removed?

File: src/Voron/StorageEnvironment.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Voron/StorageEnvironment.cs]: done

**Comment by grisha-kotler** [comment]
It was using `GenerateReport`:
https://github.com/ravendb/ravendb/blob/1817ec89a6d03b58acb4bec7c0981a4d39138a0f/src/Voron/StorageEnvironment.cs#L923

**Comment by ayende** [comment]
Wow, that was expensive

**Comment by redknightlois** [comment]
LGTM so far.

**Review by ppekrol — APPROVED** [review]

**Review by arekpalinski — APPROVED** [review]

**Comment by redknightlois** [comment]
As soon as we have the nightly built will deploy and measure again

**Comment by ppekrol** [comment]
~30min

**Pull request merged**